src/planwise/client/projects/handlers.cljs

```diff
@@ -240,33 +226,52 @@
          request-with (get-in db [:map-state :request-with])]
      (-> db
          (assoc-in [:map-state :request] (api/update-project project-id filters request-with :projects/project-updated))
-         (assoc-in [:map-state :timeout] (js/setTimeout #(dispatch [:projects/show-map-loading-sign]) 2500))
+         (assoc-in [:map-state :timeout] (js/setTimeout #(dispatch [:projects/show-map-loading-hint]) loading-hint-delay))
          (assoc-in [:map-state :current] :loading)))))
 
+(defn initiate-project-update [db request-with]
+  (let [new-db
+        (case (get-in db [:map-state :current])
+          :loaded            (loaded-to-request-pending db)
+          :request-pending   (request-pending-to-request-pending db)
+          :loading           (loading-to-request-pending db)
+          :loading-displayed (loading-displayed-to-request-pending db))]
+    (-> new-db
+        (assoc-in [:map-state :current] :request-pending)
+        (assoc-in [:map-state :request-with] request-with))))
```

Comment:
Three small details while you rebase/merge your changes:

1. instead of `-to-` use the arrow `->`, eg. `loaded->request-pending`... but:
2. all the `-to-request-pending` functions don't actually change the state to ':request-pending'; I'd go for inlining those functions here
3. I meant the vertical layout is more readable in all cases, not just `loading-to-request-pending`